app/src/main/java/com/example/boylc/demomvp/data/repository/CategoryRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.example.boylc.demomvp.data.repository;
+
+import com.example.boylc.demomvp.data.model.Category;
+import com.example.boylc.demomvp.data.source.CallBack;
+import com.example.boylc.demomvp.data.source.CategoryDataSource;
+import com.example.boylc.demomvp.data.source.remote.CategoryRemoteDataSource;
+
+import java.util.ArrayList;
+
+public class CategoryRepository implements CategoryDataSource.RemoteDataSource,CategoryDataSource.LocalDataSource {
+    private CategoryRemoteDataSource mCategoryRemoteDataSource;
```

Comment:
e nên khai báo interface thay vì class implement mới đảm bảo tính SOLID của lập trình 
```suggestion
    private CategoryDataSource. CategoryDataSource mRemoteDataSource;
```

Reply:
với những repository chỉ có local và remote e có thể remove chữ category đi vẫn đủ nghĩ, a suggest như trên